modules/dicehub/endpoints/release.go

```diff
@@ -480,12 +490,23 @@ func (e *Endpoints) DeleteReleases(ctx context.Context, r *http.Request, vars ma
 	for _, release := range releases {
 		versionList = append(versionList, release.Version)
 	}
+
 	go func() {
-		if err := e.audit(r, orgID, releasesDeleteRequest.ProjectID, identityInfo.UserID, "", "",
-			string(apistructs.BatchDeleteReleaseTemplate), map[string]interface{}{
-				"versionList": versionList,
-			}); err != nil {
-			logrus.Errorf("failed to create audit event for deleting release")
+		templateName := ""
+		auditCtx := map[string]interface{}{}
+		if len(releasesDeleteRequest.ReleaseID) == 1 {
+			templateName = string(apistructs.DeleteAppReleaseTemplate)
+			if releases[0].IsProjectRelease {
+				templateName = string(apistructs.DeleteProjectReleaseTemplate)
+			}
+			auditCtx["version"] = releases[0].Version
+			auditCtx["releaseId"] = releases[0].ReleaseID
+		} else {
+			templateName = string(apistructs.BatchDeleteReleaseTemplate)
+			auditCtx["versionList"] = strings.Join(versionList, ", ")
+		}
+		if err := e.audit(r, orgID, releasesDeleteRequest.ProjectID, identityInfo.UserID, templateName, auditCtx); err != nil {
+			logrus.Errorf("failed to create audit event for deleting release, %v", err)
```

Comment:
warp a func

Reply:
used many times

Reply:
A little duplication is often better than a little dependency.